git-workflow.md

```diff
@@ -0,0 +1,101 @@
+- `develop` - used for staging. Contains the development edge.
+- `fastlane` - used for fastlane. Contains the development edge for an epic.
+
+## Development branches
+All development work should be done in a dedicated branch. This 
+makes it easy for multiple developers to work on a particular feature without 
+disturbing the main codebase. It also means the `master` branch will never 
+contain build braking code. 
+
+## Workflow
+### Development
+When ready to start development branch out from `master`
+
+```
+git checkout master
+git pull
+git checkout -b feature/#1234-strong-passwords
+```
+
+```
+master  *-*-*
+             \
+feature/#n    \-*-*-*-*
+```
+
+### Code Review
+Once you have finished a feature, open a PR to merge into `develop`.
+
+### Acceptance
+The development branch is merged into `develop`, which should trigger a deployment
+on CircleCI.
+
+```
+master      ----*
+                 \
+feature/#n        \-*-*-*-*
+                           \
+develop     ----------------m
+```
+
+### Production
+Deployments to production are completed by creating a release branch from master and
+successively merging the selected\* feature branches into the release branch. This allows features
+to be 'cherry-picked' into production without having to wait for all features in develop
+to receive Product Owner approval. 
+
+\**Waffle is currently used to to mark issues by adding them to the "Deploying" column. Both relsr and flightplan use the
+waffle labels to identify issues for the next release.*
+
+```
+master          ---------*-------------------m-------
+                   \      \                 /
+release/20171231-101500    \----m-------m--*
+                     \         /       /
+feature/#n            \-*-*-*-*       / 
+                       \       \     / 
+feature/#x              \-*-*-*-\-*-* 
+                         \       \   \ 
+bug/#b                    \-*-*   \   \ 
+                               \   \   \ 
+develop     --------------------m---m---m----------
+```
+
+To create the release branch and PR you can either use [relsr](https://github.com/jcleary/relsr) on the command-line or allow [flightplan](https://flightplan.createl.io) to create
+them for you on the deployment schedule (currently PR at 10:15, merge at 10:30 daily).
+
+### Resolving Conflicts
+If GutHub warns you of conflicts in your PR, resolve the conflicts by merging your branch into develop on the command line. 
+Do you use the GutHub tools or the command line to merge develop into your branch as this will create deployment dependencies. 
+The process looks like this:
+```bash
+git checkout develop
+git pull
+git merge feature/#1234-strong-passwords
+# resolve conflicts
+git commit
+git push
+```
+
+When you push develop, GitHub will close the PR for you automatically. 
+
+### Rework
+1. Pull the original branch
+1. Create a rework branch without the # (e.g rework/1234-fix-breaking-test-part-1)
```

Comment:
Is this different to what we do now? I've always done rework on the same branch 😊  😱

Reply:
Yes, this is a new suggested approach to rework - open to suggestions but felt this would be nicer as the PR was just for the changes. Downside is it's an extra (manual) step to get to staging.

Reply:
I thought the same thing Rob!